webapp/app/forms/validators.py

```diff
@@ -115,3 +117,30 @@ def __call__(self, form, field):
         for char in input_str:
             if char not in VALID_UNLOCK_CODE_CHARACTERS:
                 raise ValidationError(_('validate.invalid-character'))
+
+class ValidDayOfBirth:        
+    def __call__(self, form, field):
+        raw_input_date = field._value()
+
+        if isinstance(raw_input_date, str):
+            day, month, year = raw_input_date.split('.')
+        elif isinstance(raw_input_date, list):
+            day, month, year = raw_input_date
```

Comment:
In what cases is the raw_input a dot-separated string?

Reply:
Mh, I think we in fact should probably chenge the unit tests then. Are those the unit tests explicitly setting the `.raw_data` of a field? Because we should replace those anyways.